### ISSUE TITLE
Subprocess OSError denied permissions 

### ISSUE THREAD

**Issue by 26darkcores** [kuri65536/python-for-android#128]
When I import the subprocess module and use a simple call most of the time I get an exception. That exception is OSError: permission denied. 

For example: 
subprocess.call('ls') # does not give any error
subprocess.call('ls -A') # gives an error

I'm root so I run the shell from SL4A and type "su" and grant permissions to the app. I can run any commands in it but if I start the python interpreter and type:
subprocess.call('su -c ls -A') # same error

I even run the python interpreter from terminal emulator and adb shell but I still get the same error.

This is strange.. I used the original project of SL4A and P4A but no problem arise in this matter. In fact that is how I managed to get this far as no problem stood in my way regarding permissions. 

The differences are not in my script. What only changed was:
1. stock (2.3) & custom (4.1) Inspire 4g with original SL4A & P4A
2. stock (4.4.2) Galaxy S3 with the newer debug versions provided in this page

The last exception is in _execute_child inside the subprocess module

Is it because of the android version? If so how can I get around this? Do I need to add permissions to Python for android manifest?

**Comment by kuri65536** [comment]
In my test phone, OSError happen but not security issue.

Please check subprocess.call() argument, it is a list or tuple.

"ls" => OK, "ls" file may be found.
"ls -A" => the name of "ls -A" file may not be found.

```
$ adb shell
n7000:/ # bash
n7000 / # cd /storage/sdcard0/
n7000 sdcard0 # sh python2.sh 
Python 2.7.12 (default, Jan 17 2017, 16:00:46) 
[GCC 4.8] on linux-armv7l
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess
>>> subprocess.call("ls -A")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/shimoda/python-for-android/python-build/python_arm/python-ext/python/subprocess.py", line 523, in call
  File "/home/shimoda/python-for-android/python-build/python_arm/python-ext/python/subprocess.py", line 711, in __init__
  File "/home/shimoda/python-for-android/python-build/python_arm/python-ext/python/subprocess.py", line 1343, in _execute_child
OSError: [Errno 2] No such file or directory

>>> subprocess.call("ls")
ls: ./.android_secure: Permission denied
Alarms       LOST.DIR       Pictures     Samsung        abc2.py               clockworkmod                      documents    python2.sh             
AndroSensor  Movies         Playlists    ScreenCapture  abc3.py               cloudagent                        external_sd  sl4a                   
Android      Music          Podcasts     ShareViaWifi   aps.log               com.googlecode.python3forandroid  foo.jpg      standalone_python2.sh  
Application  Notifications  PressReader  Sounds         apsdemooc-rev160.apk  com.googlecode.pythonforandroid   log          storage                
DCIM         OTAUpdates     Ringtones    Sメモ        backups               com.python3forandroid             mddata       tencent                
Download     PhotoEditor    S Note       abc.py         btsnoop_hci.log       dianxin                           media        usbStorage             
1
```

**Comment by 26darkcores** [comment]
I used "ls" as an example but "busybox ping -c 4 8.8.8.8" yields the same. 

Here are some of the commands I used:
ls -a # yeah mistyped 
ps -A # ps alone works
la
uname -a # uname alone works
busybox uname -a # busybox alone works
busybox ether-wake

Another error that rise is with "ls /", "ls /data", "ls /storage/sdcard0". All yield an OSError: [Errno2] not such file or directory and yet with call('mount') all directories are mounted and available. All binaries can be accessed in a separate shell inside SL4A, Terminal Emulator and adb shell without calling for super user.
